src/sg_client.rs

```diff
@@ -66,16 +68,39 @@ fn make_post_body(mut mail_info: Mail) -> SendgridResult<String> {
 
 impl SGClient {
     /// Makes a new SendGrid cient with the specified API key.
-    pub fn new(key: String) -> SGClient {
+    pub fn new<S: Into<String>>(key: S) -> SGClient {
+        #[cfg(feature = "async")]
+        use reqwest as rq;
+        #[cfg(not(feature = "async"))]
+        use reqwest::blocking as rq;
+
+        let builder = rq::ClientBuilder::new();
+        #[cfg(feature = "rustls")]
+        builder.use_rustls_tls();
+        let client = builder.build().unwrap(); 
```

Comment:
If this can panic, we should document that too. Or we can change the return type to be a `SendgridResult`.

Reply:
This will panic just the same as using `Client::new`, if there is no TLS backend available, only it is more explicit. I will add a comment explaining this!